pikachewie/publisher.py

```diff
@@ -51,6 +51,20 @@ def publish(self, exchange, routing_key, body, properties=None):
                 body=body,
             )
 
+    def process_data_events(self, time_limit=0):
+        """Calls process_data_events on the current connection (if there is
+        currently one open).
+        """
+        if self._channel:
+            try:
+                self.channel.connection.process_data_events(time_limit=time_limit)
+            except self.retry_on_exceptions as exc:
+                log.warn('Cannot process data events')
+
+                # If there's a problem, we'll discard the channel and reconnect
+                # it on demand.
+                self._channel = None
```

Comment:
To maintain consistent retry logic between `PublishingMixin.publish()` and `PublishingMixin.process_data_events()`, the following line is needed after line 66:

`self.channel.connection.process_data_events(time_limit=time_limit)`

This will ensure that any failed operation will be retried exactly once, and will cause `process_data_events()` to raise an exception if the retry also fails.